zpay32/invoice.go

```diff
@@ -668,6 +668,12 @@ func parseTaggedFields(invoice *Invoice, fields []byte, net *chaincfg.Params) er
 
 			invoice.FallbackAddr, err = parseFallbackAddr(base32Data, net)
 		case fieldTypeR:
+			// Check that we don't already have 20 route hints. Otherwise this
+			// represents a potential DoS vector because of pubkey parsing.
+			if len(invoice.RouteHints) == 20 {
```

Comment:
We use this constant in a couple places now, so we should extract it into a constant in this package.

Reply:
Would be nice to add a test for these cases as well.

Reply:
Added